### DISCUSSION TITLE
How to print debugging information in swc-plugin？

### DISCUSSION THREAD

**Discussion by chenwll** [swc-project/swc#9165] [Q&A] [answered]
when i run `cargo test` , println!("hello world") does not work. 
How should I print debugging information？
```
#[plugin_transform]
pub fn process_transform(program: Program, _metadata: TransformPluginProgramMetadata) -> Program {
    println!("hello world");
    program.fold_with(&mut as_folder(CatchClauseVisitor::new()))
}
```

**Comment by kdy1** [comment]
You may use `cargo test -- --nocapture`, or make your test fail. `cargo test` captures the output of tests and drop it if the test passes

**Reply by chenwll** [reply]
i use cargo test -- --nocapture，but "hello world" not print

**Reply by kdy1** [reply]
Ah, you were trying to print from Wasm plugin. You should compile it to `wasi` target if you want to use feature of std runtime.

**Reply by chenwll** [reply]
I'm a beginner in rust, can you tell me a little more about how to print out "hello world" when running a test case, this is my code
https://github.com/chenwll/swc_plugin/blob/master/src/lib.rs#L154

**Reply by kdy1** [reply, marked as answer]
`#[plugin_transform]` is not called for `test_fixture`. You should print log in the constructor function you call in the test
